src/components/rpc_base/include/rpc_base/rpc_base.h

```diff
@@ -307,6 +307,40 @@ class Array : public std::vector<T>, public CompositeType {
       policy_table_interface_base::PolicyTableType pt_type);
 };
 
+// Partial specialization of Array for Enum<E> type because of different
+// validation rules
+
+template <typename E, size_t minsize, size_t maxsize>
+class Array<Enum<E>, minsize, maxsize> : public std::vector<Enum<E> >,
+                                         public CompositeType {
+ public:
+  // Types
+  typedef std::vector<Enum<E> > ArrayType;
+
+ public:
```

Comment:
@LuxoftAKutsan redundant public modifier

Reply:
@AKalinich-Luxoft Fixed in d1308970ca2afad7de251f6ef744d94c4422cb42